resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1172,6 +1188,30 @@ private[spark] class Client(
     }.mkString("")
   }
 
+  /**
+   * Fetch links to the logs of the driver for the given application ID. This requires hitting the
+   * RM REST API. Returns an empty map if the links could not be fetched. If this feature is
+   * disabled via [[CLIENT_INCLUDE_DRIVER_LOGS_LINK]], an empty map is returned immediately.
+   */
+  private def getDriverLogsLink(appId: ApplicationId): IMap[String, String] = {
+    if (!sparkConf.get(CLIENT_INCLUDE_DRIVER_LOGS_LINK)) {
+      return IMap()
+    }
+    val baseRmUrl = WebAppUtils.getRMWebAppURLWithScheme(hadoopConf)
```

Comment:
I'd safely guard with try-catch and swallow the exception with logging - IMHO this is to add additional information, which is not major functionality to let the process fail in the middle.

Reply:
Sure, makes sense to me. Updated.